### PR TITLE
Add new struct called RMSEVariable and necessary functionalities for plotting

### LINKED ISSUE #88: Add struct and constructor for storing RMSEs across model names and seasons
In ClimaCoupler, to plot the leaderboard, the RMSEs are read from a CSV file and put into a struct to manipulate for plotting. This functionality should be moved to ClimaAnalysis and can be made to be more general.

### LINKED ISSUE #89: Add `find_best_single_model`, `find_worst_single_model`, and `median` for struct that store RMSEs
In ClimaCoupler, to plot the leaderboard, the functionality of finding the best single model, worst single model, and median model needs to be implemented. This functionality should be moved to ClimaAnalysis and can be made to be more general. For instance, finding the best/worst single model can be done over different categories rather than defaulting to "annual".

### LINKED ISSUE #90: Add `plot_box_plot` and `plot_leaderboard` using structs that store RMSEs
The plotting functionality needs to be ported over from ClimaCoupler and be compatible with the new struct holding RMSEs.

### PR DESCRIPTION
**Pull request by ph-kev**
- closes #88, closes #89, closes #90
This PR adds a new struct called `RMSEVariable` for storing root mean squared errors (RMSEs) over categories (e.g. seasons and annual) and models (e.g. CliMA). The features include for `RMSEVariable` include

- Constructors
- Reading RMSEs from CSV file
- Indexing
- Adding categories, models, and units
- Methods for summary statistics
- Plotting box plots and leaderboard

A combined plot of the box plots and leaderboard would like something below.
boxplot_and_leaderboard

### CODE REVIEW

File: src/Leaderboard.jl [hunk 1]
```diff
@@ -0,0 +1,520 @@
...
+"""
+    read_rmses(csv_file::String, short_name::String; units = nothing)
+
+Read a CSV file and create a RMSEVariable with the `short_name` of the variable.
+
+The first column should contain the model names and the rest of the columns should contain
+the root mean squared errors. The header of the CSV file should contain the category names.
+
+If units is `nothing`, then the unit is missing for each model which is denoted by an empty
+string.
```
Comment (Sbozzolo) [inline src/Leaderboard.jl:187]: We need to describe what the exepcted format for the file is

File: src/Leaderboard.jl [hunk 2]
```diff
@@ -0,0 +1,520 @@
...
+Find the median using the root mean squared errors across all categories.
+
+Any `NaN` is ignored in computing the median.
+"""
+function median(rmse_var::RMSEVariable)
+    _unit_check(rmse_var)
+    cols = eachcol(rmse_var.RMSEs)
+    median = map(cols) do col
+        no_nan_col = filter(!isnan, col)
+        median_elt = Statistics.median(no_nan_col)
```
Comment (Sbozzolo) [inline src/Leaderboard.jl:514]: We have a `NaNStatistics` package, there's probably a `nanmedian` there

File: test/sample_data/test_csv.csv [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# Model,DJF,MAM,JJA,SON,ANN
+ACCESS-CM2,11.941,10.178,13.279,10.443,8.710
+ACCESS-ESM1-5,15.752,12.477,15.955,12.972,NaN
```
Comment (Sbozzolo) [inline test/sample_data/test_csv.csv:3]: Please add new line here

File: docs/src/rmse_var.md [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+)
+rmse_var = ClimaAnalysis.RMSEVariable(
+    "ta",
+    ["ACCESS-CM2", "ACCESS-ESM1-5"],
+    ["DJF", "MAM", "JJA", "SON", "ANN"],
+    ones(2, 5),
+    "K",
+)
+
+nothing # hide
```
Comment (Sbozzolo) [inline docs/src/rmse_var.md:55]: Do you need this?

File: docs/src/rmse_var.md [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+
+```@example rmse_var
+rmse_var = ClimaAnalysis.read_rmses("./data/test_csv.csv", "ta")
+rmse_var = ClimaAnalysis.read_rmses(
+    "./data/test_csv.csv",
+    "ta",
+    units = Dict("ACCESS-CM2" => "K", "ACCESS-ESM1-5" => "K"),
+)
+
+nothing # hide
```
Comment (Sbozzolo) [inline docs/src/rmse_var.md:89]: Same here

File: docs/src/visualize_rmse_var.md [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+`model_names` are plotted. The category to find the best and worst model defaults to
+"ANN", but can be changed using the parameter `best_and_worst_category_name`.
+
+The function `plot_leaderboard!(fig, rmse_vars::ClimaAnalysis.RMSEVariable...;
+ploc = (1, 1), model_names = ["CliMA"], best_category_name = "ANN")` makes a heatmap of the
+RMSEs between the variables of interest and the categories. The best model for each variable
+of interest and the models in `model_names` are shown in the heatmap. Similar to
+`plot_boxplot!`, the category to find the best model defaults to "ANN", but can be changed
+using the parameter `best_category_name`. The values of the heatmap are normalized by
+dividing over the median model's RMSEs for each variable.
```
Comment (Sbozzolo) [inline docs/src/visualize_rmse_var.md:19]:
```suggestion
The function [`plot_boxplot!`(@ref) makes a box plot for each
category in the `RMSEVariable`. The best model and worst model and any other models in
`model_names` are plotted. The category to find the best and worst model defaults to
"ANN", but can be changed using the parameter `best_and_worst_category_name`.

The function [`plot_leaderboard!`](@ref) makes a heatmap of the
RMSEs between the variables of interest and the categories. The best model for each variable
of interest and the models in `model_names` are shown in the heatmap. Similar to
`plot_boxplot!`, the category to find the best model defaults to "ANN", but can be changed
using the parameter `best_category_name`. The values of the heatmap are normalized by
dividing over the median model's RMSEs for each variable.
```

File: ext/ClimaAnalysisMakieExt.jl [hunk 7]
```diff
@@ -630,4 +630,240 @@ function Visualize._constrained_cmap(
...
+"""
+    Visualize.plot_boxplot!(fig,
+                            rmse_var::ClimaAnalysis.RMSEVariable;
+                            model_names = ["CliMA"],
+                            ploc = (1, 1),
+                            best_and_worst_category_name = "ANN")
+
+Plot a Tukey style boxplot for each category in `rmse_var`.
+
+The best and worst single models are found for the category `best_and_worst_category_name`
```
Comment (Sbozzolo) [inline ext/ClimaAnalysisMakieExt.jl:642]:
```suggestion
The best and worst single models are found for the category [`best_and_worst_category_name`](@ref)
```

File: NEWS.md [hunk 8]
```diff
@@ -276,6 +276,119 @@ plot_bias_on_globe!(fig, sim_var, obs_var)
...
+A CSV file containing model names in the first column and root mean squared errors in the
+subsequent columns with a header describing each category (i.e. seasons) can be read into
+a `RMSEVariable`. See the example below on how to use this functionality.
+
+```julia
+rmse_var = ClimaAnalysis.read_rmses("./data/test_csv.csv", "ta")
+rmse_var = ClimaAnalysis.read_rmses(
+    "./data/test_csv.csv",
+    "ta",
+    units = Dict("ACCESS-CM2" => "K", "ACCESS-ESM1-5" => "K"), # passing units as a dictionary
```
[the remark this answers is not in the document]
Comment (ph-kev) [reply NEWS.md:337]: Now, it does

File: docs/src/rmse_var.md (hunk 4 above)
Comment (ph-kev) [reply docs/src/rmse_var.md:55]: It is a hack to make the documentation look a little nicer. If the line is removed, then we get something like this.
librewolf_isYC0OmVTn

**Pull request merged**